Proposal: Add archive.html

mkasberg opened the pull request:
Currently, it's difficult to browse old posts on opensourcecatholic.com.
We do have a search page, but that's only useful if you already know
what you're looking for. Without using the search bar, you could still
browse older articles, but there are 59 pages of articles to click
through.

Most blogs solve this problem with some kind of "archive". I.e. a way to
browse old posts by year. Some have a dedicated page, others put
something like this in a sidebar.

In this commit, I'm adding `archive.html`. It's a page that will be
listed in the top nav. The implementation isn't very long -- group all
posts by year, and list their titles in reverse chronological order.

![image](https://user-images.githubusercontent.com/12214617/95666527-b98b4e80-0b17-11eb-86b9-35b85180f585.png)

mkasberg committed:
Add archive.html

Currently, it's difficult to browse old posts on opensourcecatholic.com.
We do have a search page, but that's only useful if you already know
what you're looking for. Without using the search bar, you could still
browse older articles, but there are 59 pages of articles to click
through.

Most blogs solve this problem with some kind of "archive". I.e. a way to
browse old posts by year. Some have a dedicated page, others put
something like this in a sidebar.

In this commit, I'm adding `archive.html`. It's a page that will be
listed in the top nav. The implementation isn't very long -- group all
posts by year, and list their titles in reverse chronological order.

mkasberg commented:
Something to consider: We could combine this with the search page if we want - simply put the search bar at the top.

geerlingguy commented:
@mkasberg i think that'd be perfect; just have an Archive link, with the embedded search form at the top. Then add a redirect from search to archive. Better SEO too, having that page deeply linking into the site (hopefully)

mkasberg committed:
Combine search and archive pages

We can streamline the navigation for the site by combining the Search
and Archive pages, putting the search bar at the top of the archive
page.

The search bar loads asynchronously with JS, so placing it in a
fixed-height div prevents the page content from jumping around when the
search bar loads.

We'll redirect the `/search/` and `/search/site/` (which used to
redirect to `/search/`) paths to `/archive/`.

mkasberg commented:
:+1: Done!

![image](https://user-images.githubusercontent.com/12214617/95686209-afba2780-0bb9-11eb-8d39-c248e10d4074.png)
